src/api/db/db_model.rs

```diff
@@ -0,0 +1,189 @@
+#![allow(unused)]
+#![allow(clippy::all)]
+
+use super::schema::*;
+use diesel::prelude::*;
+use diesel::pg::PgConnection;
+use diesel::prelude::Queryable;
+use diesel::pg::data_types::PgInterval;
+use chrono::naive::{NaiveDate, NaiveDateTime};
+use serde_derive::{Deserialize, Serialize};
+use serde::ser::{Serialize, Serializer, SerializeStruct};
+use serde;
+
+#[derive(Serialize, Deserialize, Queryable, Debug, PartialEq)]
+pub struct Book {
+    pub id: i32,
+    pub isbn_13: Option<String>,
+    pub issn: Option<String>,
+    pub title: String,
+    pub subtitle: Option<String>,
+    pub description: Option<String>,
+    pub cover: Option<i32>,
+    pub edition: Option<i32>,
+    pub release_year: Option<i16>,
+    pub pages: Option<i32>,
+    pub code_identifier: Option<String>,
+    pub publisher_id: i32,
+    pub designation_id: i32,
+    pub series_id: Option<i32>,
+    pub language_id: i32,
+    pub physical_size_id: i32,
```

Comment:
Shouldn't this IDs be of type u32 instead of i32? I think postgres is capable of handling it, but it can have side-effects.

Reply:
Also subsequent lines use this data type.

Reply:
pg doesn't support unsigned integers, diesel complains about it when compiling

Reply:
Okay, alright.